chaos_genius/controllers/kpi_controller.py

```diff
@@ -167,6 +153,75 @@ def get_active_kpis() -> Iterator[Kpi]:
     return kpis
 
 
+def get_anomaly_data(
+    kpi_ids: List[int],
+    anomaly_types: List[str] = None,
+    anomalies_only: bool = False,
+    start_timestamp: datetime = None,
+    include_start_timestamp: bool = True,
+    end_timestamp: datetime = None,
+    include_end_timestamp: bool = True,
+    severity_cutoff: float = None,
+) -> List[AnomalyDataOutput]:
+    """Returns list of anomaly points using paramters to filter the output."""
+    filters = []
+    if kpi_ids:
+        filters.append(AnomalyDataOutput.kpi_id.in_(kpi_ids))
+
+    if anomaly_types:
+        # TODO: Add the series type filter for query optimisation
+        filters.append(AnomalyDataOutput.anomaly_type.in_(anomaly_types))
+
+    if anomalies_only:
+        filters.append(AnomalyDataOutput.is_anomaly.in_([1, -1]))
+
+    if start_timestamp:
+        if include_start_timestamp:
+            filters.append(AnomalyDataOutput.data_datetime >= start_timestamp)
+        else:
+            filters.append(AnomalyDataOutput.data_datetime > start_timestamp)
+
+    if end_timestamp:
+        if include_end_timestamp:
+            filters.append(AnomalyDataOutput.data_datetime <= end_timestamp)
+        else:
+            filters.append(AnomalyDataOutput.data_datetime < end_timestamp)
+
+    if severity_cutoff:
+        filters.append(AnomalyDataOutput.severity >= severity_cutoff)
+
+    anomaly_data = AnomalyDataOutput.query.filter(*filters).all()
+
+    return anomaly_data
+
+
+def get_last_anomaly_timestamp(
+    kpi_ids: List[int],
+    anomaly_types: List[str] = ["overall", "subdim"],
+) -> Optional[datetime]:
+    """Returns the timestamp of the latest anomaly data."""
+    results = (
+        AnomalyDataOutput.query.filter(
+            (AnomalyDataOutput.kpi_id.in_(kpi_ids))
+            & (AnomalyDataOutput.anomaly_type.in_(anomaly_types))
+        )
+        .order_by(AnomalyDataOutput.data_datetime.desc())
+        .first()
+    )
+
+    if results:
+        return results.data_datetime
```

Comment:
singular variable name as .first() returns either an obj or None

```suggestion
    result = (
        AnomalyDataOutput.query.filter(
            (AnomalyDataOutput.kpi_id.in_(kpi_ids))
            & (AnomalyDataOutput.anomaly_type.in_(anomaly_types))
        )
        .order_by(AnomalyDataOutput.data_datetime.desc())
        .first()
    )

    if result:
        return result.data_datetime
```

Reply:
Good catch